metastore/if/hive_metastore.thrift

```diff
@@ -1127,6 +1127,7 @@ service ThriftHiveMetastore extends fb303.FacebookService
                        throws(1:NoSuchObjectException o1, 2:MetaException o3)
   void truncate_table(1:string dbName, 2:string tableName, 3:list<string> partNames)
                           throws(1:MetaException o1)
+  void cm_recycle(1:string dataPath, 2:bool isCopy, 3:bool isPurge) throws(1:MetaException o1)
```

Comment:
The recommended style for new metastore api is to use a single request argument for the functions.
Please see discussion in http://search-hadoop.com/m/Hive/h4wBF1Ml9wxmO9z4?subj=Thoughts+on+new+metastore+APIs